### ISSUE TITLE
Packer Crash interface conversion: types.AnyType is nil, not types.ManagedObjectReference

### ISSUE THREAD

**Issue by tman5** [hashicorp/packer#8983]
Here is the crash log

```2020/04/01 15:03:36 [INFO] Packer version: 1.5.4 [go1.13.7 windows amd64]
2020/04/01 15:03:36 [DEBUG] Discovered plugin: vsphere-clone = C:\packer_1.5.4_windows_amd64\packer-builder-vsphere-clone.exe
2020/04/01 15:03:36 using external builders [vsphere-clone]
2020/04/01 15:03:36 [DEBUG] Discovered plugin: windows-update = C:\packer_1.5.4_windows_amd64\packer-provisioner-windows-update.exe
2020/04/01 15:03:36 using external provisioners [windows-update]
2020/04/01 15:03:36 Checking 'PACKER_CONFIG' for a config file path
2020/04/01 15:03:36 'PACKER_CONFIG' not set; checking the default config file path
2020/04/01 15:03:36 Attempting to open config file: C:\Users\user1\AppData\Roaming\packer.config
2020/04/01 15:03:36 [WARN] Config file doesn't exist: C:\Users\user1\AppData\Roaming\packer.config
2020/04/01 15:03:36 Setting cache directory: C:\packer_1.5.4_windows_amd64\setup\packer_cache
cannot determine if process is in background: Process background check error: not implemented yet
2020/04/01 15:03:36 Creating plugin client for path: C:\packer_1.5.4_windows_amd64\packer.exe
2020/04/01 15:03:36 Starting plugin: C:\packer_1.5.4_windows_amd64\packer.exe []string{"C:\\packer_1.5.4_windows_amd64\\packer.exe", "plugin", "packer-builder-vsphere-iso"}
2020/04/01 15:03:37 Waiting for RPC address for: C:\packer_1.5.4_windows_amd64\packer.exe
2020/04/01 15:03:37 packer.exe plugin: [INFO] Packer version: 1.5.4 [go1.13.7 windows amd64]
2020/04/01 15:03:37 packer.exe plugin: Checking 'PACKER_CONFIG' for a config file path
2020/04/01 15:03:37 packer.exe plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/04/01 15:03:37 packer.exe plugin: Attempting to open config file: C:\Users\user1\AppData\Roaming\packer.config
2020/04/01 15:03:37 packer.exe plugin: [WARN] Config file doesn't exist: C:\Users\user1\AppData\Roaming\packer.config
2020/04/01 15:03:37 packer.exe plugin: Setting cache directory: C:\packer_1.5.4_windows_amd64\setup\packer_cache
2020/04/01 15:03:37 packer.exe plugin: args: []string{"packer-builder-vsphere-iso"}
2020/04/01 15:03:37 packer.exe plugin: Plugin port range: [10000,25000]
2020/04/01 15:03:37 packer.exe plugin: Plugin address: tcp 127.0.0.1:10000
2020/04/01 15:03:37 packer.exe plugin: Waiting for connection...
2020/04/01 15:03:37 Received tcp RPC address for C:\packer_1.5.4_windows_amd64\packer.exe: addr is 127.0.0.1:10000
2020/04/01 15:03:37 packer.exe plugin: Serving a plugin connection...
2020/04/01 15:03:37 ui: vsphere-iso: output will be in this color.
2020/04/01 15:03:37 ui: 
2020/04/01 15:03:37 Build debug mode: false
2020/04/01 15:03:37 Force build: false
2020/04/01 15:03:37 On error: 
2020/04/01 15:03:37 Preparing build: vsphere-iso
2020/04/01 15:03:37 Waiting on builds to complete...
2020/04/01 15:03:37 Starting build run: vsphere-iso
2020/04/01 15:03:37 Running builder: vsphere-iso
2020/04/01 15:03:37 [INFO] (telemetry) Starting builder vsphere-iso
2020/04/01 15:03:38 ui: ==> vsphere-iso: Creating VM...
2020/04/01 15:03:40 packer.exe plugin: panic: interface conversion: types.AnyType is nil, not types.ManagedObjectReference
2020/04/01 15:03:40 packer.exe plugin: 
2020/04/01 15:03:40 packer.exe plugin: goroutine 23 [running]:
2020/04/01 15:03:40 packer.exe plugin: github.com/hashicorp/packer/builder/vsphere/driver.(*Driver).CreateVM(0xc0001e42a0, 0xc00055f668, 0x0, 0x0, 0x1)
2020/04/01 15:03:40 packer.exe plugin: 	/Users/mmarsh/Projects/packer/builder/vsphere/driver/vm.go:165 +0xbca
2020/04/01 15:03:40 packer.exe plugin: github.com/hashicorp/packer/builder/vsphere/iso.(*StepCreateVM).Run(0xc000132560, 0x496a080, 0xc0003da040, 0x4947140, 0xc00044c810, 0x0)
2020/04/01 15:03:40 packer.exe plugin: 	/Users/mmarsh/Projects/packer/builder/vsphere/iso/step_create.go:116 +0x4f5
2020/04/01 15:03:40 packer.exe plugin: github.com/hashicorp/packer/helper/multistep.(*BasicRunner).Run(0xc00044c9c0, 0x496a080, 0xc0003da040, 0x4947140, 0xc00044c810)
2020/04/01 15:03:40 packer.exe plugin: 	/Users/mmarsh/Projects/packer/helper/multistep/basic_runner.go:67 +0x21e
2020/04/01 15:03:40 packer.exe plugin: github.com/hashicorp/packer/builder/vsphere/iso.(*Builder).Run(0xc000178600, 0x496a080, 0xc0003da040, 0x49849a0, 0xc00044c7b0, 0x48f41c0, 0xc000132540, 0x2030000, 0x20, 0xc000046088, ...)
2020/04/01 15:03:40 packer.exe plugin: 	/Users/mmarsh/Projects/packer/builder/vsphere/iso/builder.go:136 +0xc22
2020/04/01 15:03:40 packer.exe plugin: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc0004385c0, 0x1, 0xc000046080, 0x0, 0x0)
2020/04/01 15:03:40 packer.exe plugin: 	/Users/mmarsh/Projects/packer/packer/rpc/builder.go:117 +0x283
2020/04/01 15:03:40 packer.exe plugin: reflect.Value.call(0xc000430660, 0xc0000060f8, 0x13, 0x42553e8, 0x4, 0xc0005a1f18, 0x3, 0x3, 0xc0005a1e78, 0x405753, ...)
2020/04/01 15:03:40 packer.exe plugin: 	/usr/local/go/src/reflect/value.go:460 +0x5fd
2020/04/01 15:03:40 packer.exe plugin: reflect.Value.Call(0xc000430660, 0xc0000060f8, 0x13, 0xc0005a1f18, 0x3, 0x3, 0xc0003ca180, 0xc0000e3500, 0xc0000e3588)
2020/04/01 15:03:40 packer.exe plugin: 	/usr/local/go/src/reflect/value.go:321 +0xbb
2020/04/01 15:03:40 packer.exe plugin: net/rpc.(*service).call(0xc000438600, 0xc00035e230, 0xc00045d208, 0xc00045d210, 0xc0000d6680, 0xc00045f0a0, 0x3605c00, 0xc00004607c, 0x18a, 0x3542420, ...)
2020/04/01 15:03:40 packer.exe plugin: 	/usr/local/go/src/net/rpc/server.go:377 +0x176
2020/04/01 15:03:40 packer.exe plugin: created by net/rpc.(*Server).ServeCodec
2020/04/01 15:03:40 packer.exe plugin: 	/usr/local/go/src/net/rpc/server.go:474 +0x432
2020/04/01 15:03:40 [INFO] (telemetry) ending vsphere-iso
2020/04/01 15:03:40 ui error: Build 'vsphere-iso' errored: unexpected EOF
2020/04/01 15:03:40 machine readable: error-count []string{"1"}
2020/04/01 15:03:40 ui error: 
==> Some builds didn't complete successfully and had errors:
2020/04/01 15:03:40 machine readable: vsphere-iso,error []string{"unexpected EOF"}
2020/04/01 15:03:40 ui error: --> vsphere-iso: unexpected EOF
2020/04/01 15:03:40 ui: 
==> Builds finished but no artifacts were created.
2020/04/01 15:03:40 Cancelling builder after context cancellation context canceled
2020/04/01 15:03:40 Error cancelling builder: connection is shut down
2020/04/01 15:03:40 [INFO] (telemetry) Finalizing.
2020/04/01 15:03:40 C:\packer_1.5.4_windows_amd64\packer.exe: plugin process exited
2020/04/01 15:03:40 waiting for all plugin processes to complete...
```

Here is my JSON
```
{
  "builders": [
    {
      "type": "vsphere-iso",

       "vcenter_server": "vcenter.testdomain.local",
        "insecure_connection": "true",
        "username": "packertest@vsphere.local",
        "password": "testpw",
        "cluster": "Compute01",
        "host": "esx1.testdomain.local",

     "vm_name":  "packerimage",
		"folder": "Templates",

      "guest_os_type": "windows9_64Guest",

      "communicator": "winrm",
      "winrm_username": "jetbrains",
      "winrm_password": "jetbrains",

      "CPUs":             1,
      "RAM":              4096,
      "RAM_reserve_all": true,

      "disk_controller_type":  "pvscsi",
      "disk_size":        32768,
      "disk_thin_provisioned": false,
	  "datastore": "Content Library",

      "network_card": "vmxnet3",
	"iso_paths": [
        "[Content LIbrary] ISO/Windows/Windows_Server_2016.ISO",
        "[Content LIbrary] ISO/Windows/VMwareTools.iso"
      ],
      "floppy_files": [
        "setup/autounattend.xml",
        "setup/setup.ps1",
        "setup/vmtools.cmd"
      ]
        
      }
    ]
  }
```

**Comment by nywilken** [comment]
Hi thanks for reaching out. Quickly looking at your config and looking at where the panic is coming from it seems like there might be an issue when trying to find the `datastore` associated with the cluster. I'll try to dive in a little deeper to see what's going on when I have a chance. 

in the meantime, If you explicitly set the [datastore](https://packer.io/docs/builders/vsphere-iso.html#datastore)  option within the configuration for your cluster. Do you get the same issue?

**Comment by tman5** [comment]
Thanks. I am specifying a datastore in my config `datastore "Content LIbrary"`

I should also note that the VM is actually getting created in the cluster on the correct host and datastore. The VM has the correct network and disk size, but that is where it stops. It never configures the CPU, memory, ISOs, OS, etc. So it's correctly placing the VM shell but never fully creates it.

**Comment by jhawk28** [comment]
"Content Library" is not the same as [Content LIbrary]. Does the upper case 'I' matter?

**Comment by jhawk28** [comment]
just a note that govc is expecting that the taskInfo.Result is also a ManagedObjectReference: https://github.com/vmware/govmomi/blob/master/govc/vm/create.go#L320

**Comment by tman5** [comment]
The datastore did have a capital I in it, but it was the same in the JSON file too. I renamed the datastore with a lowercase i just to be safe and updated my JSON with the updated name. Same crash.

**Comment by sylviamoss** [comment]
Hey @tman5, I'm working on this issue trying to figure out what is going on. This error occurs because the result of creating the VM is nil and it doesn't return any error. Packer will then try to work on that result, which is nil, and this is causing the crash. 
Right now there's no way to log what exactly went wrong from Packer side. Are there any logs in your cluster that you could check if something happened when creating the VM? 

CPU, memory, and etc... are not configured because these are steps that Packer runs after the VM creation and the build is failing before that.

**Comment by sylviamoss** [comment]
Since the crash is fixed by https://github.com/hashicorp/packer/pull/9354 I'm going to close this issue for now and if you find another issue when running the build, please let us know.
